core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -596,18 +596,24 @@ object Chunk
       case ix: GIndexedSeq[O] => indexedSeq(ix)
       case _ =>
         if (i.isEmpty) empty
-        else {
-          val itr = i.iterator
-          val head = itr.next()
-          if (itr.hasNext) {
-            val bldr = collection.mutable.Buffer.newBuilder[O]
-            bldr += head
-            bldr ++= itr
-            buffer(bldr.result())
-          } else singleton(head)
-        }
+        else iterableOnce(i)
     })
 
+  /** Creates a chunk from a `scala.collection.IterableOnce`. */
+  def iterableOnce[O](i: collection.IterableOnce[O]): Chunk[O] = {
+    val itr = i.iterator
+    if (itr.isEmpty) empty
+    else {
+      val head = itr.next()
+      if (itr.hasNext) {
+        val bldr = collection.mutable.Buffer.newBuilder[O]
```

Comment:
According to this
https://github.com/typelevel/fs2/pull/2720/files#diff-e8fdf3ccd3d2479432355b3abf34a16cd548373afeddeaf231b8d8a1147b1d48R115
ArrayBuilder shows x2 performance compared to mutable.Buffer, shouldn't we change a type of builder used here?

Reply:
In those cases we have the (expected) size, we don't in this case.

Reply:
ArrayBuilder should still beat mutable buffer though, even without a size hint.

Reply:
Using `ArrayBuilder` requires us to add a `ClassTag` constraint, I don't think that is something that we want?

That would mean that we couldn't use it in `iterable` and `chain`.

Reply:
Merge the main branch in to this PR and then use `makeClassTag[Any]`. We'll have to cast the resulting chunk back to a `Chunk[A]` just like we do in methods like `filter`.